Add recipe for gdb-x

Grimpper opened the pull request:
### Brief summary of what the package does

 Extra utilities for improving the user interface of gdb-mi.el.

### Direct link to the package repository

https://codeberg.org/shepherd/gdb-x

### Your association with the package

Maintainer

### Relevant communications with the upstream package maintainer

**None needed** 

### Checklist

<!-- Please confirm by replacing `[]` with `[x]`: -->

- [x] The package is released under a [GPL-Compatible Free Software License](https://www.gnu.org/licenses/license-list.en.html#GPLCompatibleLicenses)
- [x] I've read [CONTRIBUTING.org](https://github.com/melpa/melpa/blob/master/CONTRIBUTING.org)
- [x] I've used the latest version of [package-lint](https://github.com/purcell/package-lint) to check for packaging issues, and addressed its feedback
- [x] My elisp byte-compiles cleanly
- [x] I've used `M-x checkdoc` to check the package's documentation strings
- [x] I've built and installed the package using the instructions in [CONTRIBUTING.org](https://github.com/melpa/melpa/blob/master/CONTRIBUTING.org)

<!-- After submitting, please fix any problems the CI reports. -->

riscy commented:
Thanks for this.  A couple quick comments --

This package is advising 4 functions at load time.
- In two cases this advise is defined in terms of lambdas; I'd suggest named functions for these, so that it's easier to get documentation to users about how/why their function has been advised
- Heaping advice onto the runtime at load time is something I (gently) push back against; are these runtime modifications that we can control with a global minor mode?  Another consideration when you're using advice (regardless of whether you package it in a togglable minor mode) is that I believe if a user runs `M-x unload-feature` on your package, the advice remains, and this can cause errors unless you write a `gdb-x-unload-function` function.

Sharp-quote stuff, fyi:
- gdb-x.el#L83: There is no need to quote lambdas (neither #' nor ')
- gdb-x.el#L213: There is no need to quote lambdas (neither #' nor ')
- gdb-x.el#L227: There is no need to quote lambdas (neither #' nor ')
- gdb-x.el#L281: There is no need to quote lambdas (neither #' nor ')
- gdb-x.el#L287: There is no need to quote lambdas (neither #' nor ')

Grimpper commented:
>     * In two cases this advise is defined in terms of lambdas; I'd suggest named functions for these, so that it's easier to get documentation to users about how/why their function has been advised

Thanks for letting me know. I've replaced the lambdas for named functions.

>     * Heaping advice onto the runtime at load time is something I (gently) push back against; are these runtime modifications that we can control with a global minor mode?  Another consideration when you're using advice (regardless of whether you package it in a togglable minor mode) is that I believe if a user runs `M-x unload-feature` on your package, the advice remains, and this can cause errors unless you write a `gdb-x-unload-function` function.

That's good to know. I've moved all top level advice to `gdb-x-mode` and added `gdb-x-unload-function`, which disables the modes if enabled. Disabling the modes removes all the advice added by the package.

> Sharp-quote stuff, fyi:
> 
>     * gdb-x.el#L83: There is no need to quote lambdas (neither #' nor ')
> 
>     * gdb-x.el#L213: There is no need to quote lambdas (neither #' nor ')
> 
>     * gdb-x.el#L227: There is no need to quote lambdas (neither #' nor ')
> 
>     * gdb-x.el#L281: There is no need to quote lambdas (neither #' nor ')
> 
>     * gdb-x.el#L287: There is no need to quote lambdas (neither #' nor ')

I know it's not compulsorily but I like to put them for the shake of consistency when referencing functions. If it's okay, I'd rather keep the sharp quotes.

Let me know if I should make any further correction to get the package recipe merged.
Have a good day!
Sergio.

riscy commented:
Thank you!  Yes, the sharp-quoted lambdas are absolutely fine. :)

And, welcome to MELPA!
